### PR TITLE
Add sendQueryWithHeaders, sendMutationWithHeaders to support Authorization tokens

### PR DESCRIPTION
**Pull request by gothy**
`BaaS`es like graph.cool and scaphold.io are using http headers to authorize graphql requests.
I had to add this custom functions to pass the authorization token. 
I hope my code is OK, I've tried to format some of it up to `elm-format` standards, but not sure if everything is absolutely fine.
API changes in `Http/Client` are not breaking, so this could be a minor version update :)

### CODE REVIEW

**Comment by jamesmacaulay** [comment]
Good call 👍 this was an oversight on my part to leave this out. In fact this is probably such a common requirement that it might be worth just adding the extra arg to the existing functions instead of adding new ones. It would be a major version bump, but maybe preferable to having two extra functions... 🤔

**Comment by gothy** [comment]
Being a fan of Rich Hickey I have to give you link to this talk https://youtu.be/oyLBGkS5ICk?t=19m51s :) The whole talk is super-interesting, imho, but I've skipped a bit to the topic.

tldr; It's always better to "grow" your api than break it. And "growing" means adding new stuff without changing old stuff that already there and works for somebody.
That's why I've made another two functions instead of changing the original ones :)

**Comment by lpil** [comment]
What about changing the HTTP method? Or the timeout? What if we want to send with credentials?

How about instead of this we take the approach that `Http` takes and provide one high level "this is probably what you want" type method, and another one (i.e. `Http.request`) that takes a record where the user can specify all the configuration.

It seems that for this library we would want something like this:


```elm
type alias GraphQLHttpRequest =
    { method : String
    , headers : List Http.Headers
    , url : String
    , timeout : Maybe Int
    , withCredentials : Bool
    }
```

**Comment by gothy** [comment]
@lpil absolutely. That was also one of my concerns when I was preparing this PR 😄 
But I'm not a "pro" in Elm yet, so I might not see the best way to design this API. I was just stuck with my problem with headers so I've decided to offer the simplest solution to this particular issue. Seems like I'm not alone 😄

**Comment by jamesmacaulay** [comment]
Closed in favour of #7. Thanks for the input!
